File: java/server/src/main/java/com/github/fernthedev/lightchat/server/api/IPacketHandler.java
package com.github.fernthedev.server.api;

import com.github.fernthedev.lightchat.core.packets.Packet;
import com.github.fernthedev.server.ClientConnection;

@FunctionalInterface
public interface IPacketHandler {

    void handlePacket(Packet packet, ClientConnection clientConnection, int packetId);

}

File: java/server/src/main/java/com/github/fernthedev/lightchat/server/api/IPacketHandler.kt
package com.github.fernthedev.lightchat.server.api

import com.github.fernthedev.lightchat.core.codecs.AcceptablePacketTypes
import com.github.fernthedev.lightchat.server.ClientConnection

fun interface IPacketHandler {
    suspend fun handlePacket(acceptablePacketTypes: AcceptablePacketTypes, clientConnection: ClientConnection, packetId: Int)
}